src/sentry/api/serializers/models/incident.py

```diff
@@ -95,3 +100,12 @@ def _build_discover_query(self, incident):
             incident.alert_rule.snuba_query.event_types,
             discover=True,
         )
+
+
+class DetailedIncidentActivitySerializer(DetailedIncidentSerializer):
+    def serialize(self, obj, attrs, user):
+        context = super().serialize(obj, attrs, user)
+        context["activities"] = serialize(
+            list(IncidentActivity.objects.filter(incident=obj).select_related("user", "incident"))
+        )
```

Comment:
This will need to happen in bulk in `get_attrs`, otherwise we're making a query for every incident.